src/pages/[collectionSlug]/index.astro

```diff
@@ -0,0 +1,136 @@
+---
+import CollectionItemList from "@/components/CollectionItemList.astro";
+import PagesSection from "@/components/PagesSection.astro";
+import Layout from "@/layouts/Layout.astro";
+import PaginationNav from "@/components/PaginationNav.astro";
+import PageWithSideNav from "@/components/PageWithSideNav.astro";
+import RichText from "@/components/RichText.astro";
+import Media from "@/components/Media.astro";
+import {
+  fetchCustomCollectionPages,
+  createCustomCollectionPageMapper,
+  fetchCollection,
+  fetchSlug,
+} from "@/utilities/fetch";
+import { paginate } from "@/utilities/pagination";
+import { safeJsonParse } from "@/utilities/fetch/payload-fetch";
+
+export const PAGE_SIZE = import.meta.env.PAGE_SIZE || 10;
+
+const { collectionSlug } = Astro.params;
+
+// Check if this slug matches a custom collection config
+const collectionConfig = await fetchSlug(
+  "custom-collections",
+  collectionSlug || "",
+);
+
+let page = null;
+let collectionPages: any[] = [];
+let items: any[] = [];
+let currentPage = 1;
+let totalPages = 1;
+let hasPaginationNav = false;
+
+if (!collectionConfig) {
+  // Check if this is a regular page - if so, render it here since this route matched first
+  page = await fetchSlug("pages", collectionSlug || "");
+
+  if (!page) {
+    return Astro.redirect("/404");
+  }
+} else {
+  // Fetch pages for this custom collection
+  const collectionPagesData = await fetchCustomCollectionPages(
+    collectionConfig.id,
+  );
+  collectionPages = collectionPagesData?.docs || [];
+
+  // Sort by publishedAt
+  const sorted = collectionPages.sort((a: any, b: any) => {
+    return (
+      new Date(b.publishedAt || 0).getTime() -
+      new Date(a.publishedAt || 0).getTime()
+    );
+  });
+
+  // Paginate
+  currentPage = 1;
+  hasPaginationNav = sorted.length >= PAGE_SIZE;
+  const paginationResult = paginate(sorted, currentPage, PAGE_SIZE);
+  totalPages = paginationResult.totalPages;
+  const paginatedItems = paginationResult.paginatedItems;
+
+  // Map items
+  const mapper = createCustomCollectionPageMapper(collectionSlug || "");
+  items = paginatedItems.map(mapper);
+}
+
+export async function getStaticPaths() {
+  // Fetch all custom collection configs to generate static paths
+  try {
+    const response = await fetchCollection("custom-collections");
+    const data = await safeJsonParse(response);
```

Comment:
Delete and rename the `response` above to `data` since it already gets called in the `fetchCollection`